ansible_builder/main.py

```diff
@@ -71,13 +118,93 @@ def version(self):
+
     @property
-    def galaxy_requirements_file(self):
-        galaxy_file = self.raw.get('dependencies', {}).get('galaxy')
-        if galaxy_file is None or os.path.isabs(galaxy_file):
-            return galaxy_file
+    def python_requirements_file(self):
+        return self._get_dep_entry('python')
+
+    @property
+    def system_requirements_file(self):
+        return self._get_dep_entry('system')
+
+
+class CollectionDefinition(BaseDefinition):
+    """This class represents the dependency metadata for a collection
+    should be replaced by logic to hit the Galaxy API if made available
+    """
+
+    def __init__(self, collection_path):
+        super(CollectionDefinition, self).__init__(collection_path)
+        meta_file = os.path.join(collection_path, 'meta', constants.default_file)
+        if os.path.exists(meta_file):
+            with open(meta_file, 'r') as f:
+                self.raw = yaml.load(f)
         else:
-            return os.path.join(os.path.dirname(self.filename), galaxy_file)
+            # A feature? Automatically infer requirements for collection
+            for entry, filename in [('python', 'requirements.txt'), ('system', 'bindep.txt')]:
+                candidate_file = os.path.join(collection_path, filename)
+                if os.path.exists(candidate_file):
+                    self.raw['dependencies'][entry] = candidate_file
+
+    def target_dir(self):
+        namespace, name = self.namespace_name()
+        return os.path.join(
+            constants.base_collections_path, 'ansible_collections',
+            namespace, name
+        )
+
+    def namespace_name(self):
+        "Returns 2-tuple of namespace and name"
+        path_parts = [p for p in self.reference_path.split(os.path.sep) if p]
+        return tuple(path_parts[-2:])
+
+
+class UserDefinition(BaseDefinition):
+    def __init__(self, filename):
+        self.filename = filename
+        self.reference_path = os.path.dirname(filename)
+        self._manager = None
+
+        try:
+            with open(filename, 'r') as f:
+                self.raw = yaml.load(f)
+        except FileNotFoundError:
+            sys.exit("""
+            Could not detect '{0}' file in this directory.
+            Use -f to specify a different location.
+            """.format(constants.default_file))
+
+    @property
+    def galaxy_requirements_file(self):
+        return self._get_dep_entry('galaxy')
+
+    def collection_dependencies(self):
+        ret = {'python': [], 'system': []}
+        if not self.manager:
+            return
+        for path in self.manager.path_list():
+            CD = CollectionDefinition(path)
+            for entry, f in [('python', CD.python_requirements_file), ('system', CD.system_requirements_file)]:
+                if not f:
+                    continue
+                f_name = os.path.basename(f)
+                ret[entry].append(os.path.join(CD.target_dir(), f_name))
```

Comment:
This is something I had misgivings about. This method returns a list of paths inside of `/usr/share/ansible`, meaning, absolute paths in the container. I don't like that, because we would like the `Definition` classes to only be concerned with relative paths inside the project.

However, I also don't want the `GalaxySteps` class to initialize any `CollectionDefinition`. Maaaybe we could do something nicer by running the install command inside the root directory of collections, or by passing the base path to this method from `GalaxySteps`.